Framework/Core/src/ArrowTableSlicingCache.cxx

```diff
@@ -25,6 +25,14 @@ std::pair<int64_t, int64_t> SliceInfoPtr::getSliceFor(int value) const
   if (values.empty()) {
     return {offset, 0};
   }
+  auto p = values.size() - 1;
+  while (values[p] < 0) {
```

Comment:
All-negative values of indices would indicate a malformed table. I can add a check for that, but it should be on cache creation, rather then here.

Reply:
The assert is there exactly to indicate preconditions which the reader might or might not know about. I think it's good practice to have those kind of asserts, so that one does not have to keep into account the inner details in order to review the PR.